rust/src/livefs.rs

```diff
@@ -443,57 +467,14 @@ mod test {
     }
 }
 
-mod ffi {
-    use super::*;
-    use glib;
-    use glib::translate::*;
-    use glib::GString;
-    use glib_sys;
-    use libc;
-
-    use crate::ffiutil::*;
-
-    #[no_mangle]
-    pub extern "C" fn ror_livefs_get_state(
-        sysroot: *mut ostree_sys::OstreeSysroot,
-        deployment: *mut ostree_sys::OstreeDeployment,
-        out_inprogress: *mut *mut libc::c_char,
-        out_replaced: *mut *mut libc::c_char,
-        gerror: *mut *mut glib_sys::GError,
-    ) -> libc::c_int {
-        let _sysroot: ostree::Sysroot = unsafe { from_glib_none(sysroot) };
-        let deployment: ostree::Deployment = unsafe { from_glib_none(deployment) };
-        match get_livefs_state(&deployment) {
-            Ok(Some(state)) => {
-                unsafe {
-                    if let Some(c) = state.inprogress_commit {
-                        *out_inprogress = c.to_glib_full();
-                    }
-                    if let Some(c) = state.commit {
-                        *out_replaced = c.to_glib_full();
-                    }
-                }
-                1
-            }
-            Ok(None) => 1,
-            Err(ref e) => {
-                error_to_glib(e, gerror);
-                0
-            }
-        }
```

Comment:
Deleting the code above and not having to write anything like it again is I think what's going to make all this cxx-rs effort worth it.